api_docgen/Makefile.docfiles

```diff
@@ -42,11 +42,20 @@ STDLIB=$(filter-out stdlib__Pervasives, $(STDLIB_MODULES))
 
 stdlib_UNPREFIXED=$(filter-out pervasives, $(STDLIB_MODULE_BASENAMES))
 
-otherlibref= \
-  $(str_MLIS:%.mli=%) \
-  $(unix_MLIS:%.mli=%) \
-  $(dynlink_MLIS:%.mli=%) \
-  $(thread_MLIS:%.mli=%)
+otherlibref := $(dynlink_MLIS:%.mli=%)
+
+ifneq "$(filter str,$(OTHERLIBRARIES))" ""
+otherlibref += $(str_MLIS:%.mli=%)
+endif
+
+ifneq "$(filter %unix,$(OTHERLIBRARIES))" ""
+otherlibref += $(unix_MLIS:%.mli=%)
+endif
+
+ifneq "$(filter systhreads,$(OTHERLIBRARIES))" ""
+otherlibref += $(thread_MLIS:%.mli=%)
```

Comment:
If I am not mistaken, this is the only change which is really observable, since in all other cases `ocamldoc` will fail to build.

Reply:
I hadn't thought of that! However, it'll become slightly more relevant when the switch to odoc is completed, I guess?